graph/src/components/link_resolver.rs

```diff
@@ -18,6 +19,9 @@ pub type JsonValueStream =
 
 /// Resolves links to subgraph manifests and resources referenced by them.
 pub trait LinkResolver: Send + Sync + 'static + Sized {
+    /// Creates a new link resolver that implements the given timeout.
+    fn with_timeout(&self, timeout: Duration) -> Box<Self>;
```

Comment:
I'd think this `Box` could be avoided since we have a `Sized` bound on the trait.

Reply:
I tried just `Self` but it didn't compile.

Reply:
Compiles now, I'll push an update.